agent/engine/docker_task_engine_linux.go

```diff
@@ -39,3 +44,78 @@ func (engine *DockerTaskEngine) updateTaskENIDependencies(task *apitask.Task) {
 func (engine *DockerTaskEngine) invokePluginsForContainer(task *apitask.Task, container *apicontainer.Container) error {
 	return nil
 }
+
+func (engine *DockerTaskEngine) watchAppNetImage(ctx context.Context, filepath string) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		seelog.Errorf("error adding %s to fsnotify watcher, error: %v", filepath, err)
+	}
+	err = watcher.Add(filepath)
+	if err != nil {
+		seelog.Errorf("failed to initialize fsnotify NewWatcher, error: %v", err)
+	}
+	defer watcher.Close()
+	engine.watch(ctx, watcher)
+}
+
+func (engine *DockerTaskEngine) watch(ctx context.Context, watcher *fsnotify.Watcher) {
```

Comment:
So unit test failed  with `open /tmp/serviceconnect/agent.tar : permission denied` in previous revision when we tried to test it by creating a file in `/tmp`, my understanding is that we need to run this test with sudo permission in order for test to create files with os package(this test passed on local when run with sudo command).

I split the watchAppnetImage into two functions so that we could test the image reload and relay task restart logic by creating fake event and avoid creating any temp files.

Reply:
We should be able to use `TempDir()` as the base dir, instead of `/tmp` which requires privileged access.
https://pkg.go.dev/testing#T.TempDir

Reply:
Thank you, `TempDir()` worked:)